### ISSUE TITLE
Error when running Liquibase: Table already exists

### ISSUE THREAD

**Issue by phxql** [liquibase/liquibase-cache#1]
Hello,

i'm getting an error when running Liquibase against a Cache database:

```
>liquibase status
Unexpected error running Liquibase: [SQLCODE: <-201>:<Table or view name not unique>]
[Location: <ServerLoop>]
[%msg: <Table 'SQLUser.DATABASECHANGELOG' already exists>]
```

When the database is empty, the first update works:
```
>liquibase update
Liquibase Update Successful
```

But every subsequent update call fails:
```
C:\Users\moritz.kammerer\Downloads\liquibase-3.3.2-bin>liquibase update
Unexpected error running Liquibase: java.sql.SQLException: [SQLCODE: <-201>:<Table or view name not unique>]
[Location: <ServerLoop>]
[%msg: <Table 'SQLUser.DATABASECHANGELOGLOCK' already exists>]
```

**Comment by harish-myaccount** [comment]
I am getting this error when I try to run update second time

I tried <defaultSchemaName> tag in configuration as suggested in some forum.No use.

Failed to execute goal org.liquibase:**liquibase-maven-plugin:3.4.2:update** (default) on project idp-external-auth: Error setting up or running Liquibase: Table 'DATABASECHANGELOG' already exists [Failed SQL: CREATE TABLE mysql.DATABASECHANGELOG (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED datetime NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35) NULL, DESCRIPTION VARCHAR(255) NULL, COMMENTS VARCHAR(255) NULL, TAG VARCHAR(255) NULL, LIQUIBASE VARCHAR(20) NULL, CONTEXTS VARCHAR(255) NULL, LABELS VARCHAR(255) NULL)]

**Comment by onkar5891** [comment]
Hello,
I had same problem when I used MySQL. The problem was 'mysql' schema does not recognise 'DATABASECHANGELOGLOCK' as it treats its lower case name differently. So, I changed the variable 'lower_case_table_names' to 1 [But as this variable is read only, we need to change in mysqld.cnf], restarted the mysql service and then, subsequent update of liquibase started working.

Have a look at similar variable in your cache database so that 'DATABASECHANGELOGLOCK' is treated in the same way regardless of the case.

Thanks,
Onkar

**Comment by molivasdat** [comment]
Hi @phxql .  Are you still using this Liquibase Cache database extension?
How about @onkar5891?
@elktmamer @ericfrancis @kurbatov @vivekmore?
Any feedback?

**Comment by kurbatov** [comment]
I had the same error when using H2 database with `database_to_upper=false` parameter in URL.

I got around my problem by monkey-patching `liquibase.database.core.H2Database` class. Namely, I commented out the first line of the constructor which reads:

```java
super.unquotedObjectsAreUppercased=true;
```

Before I did that I researched sources of liquibase. I found out that when it checks if DATABASECHANGELOGLOCK table exists it does it in a different way than checking DATABASECHANGELOG for existence. I don't remember exactly what the difference was but probably it checks for uppercased name of the table adding quotes nonetheless which leads to a negative result and subsequent attempt to create DATABASECHANGELOGLOCK again.

How I did monkey-patching: I created a package `org.liquibase.database.core` in my project and copied `H2Database` modifying it there. When a classloader is looking for that class, my version gets priority because my artifact is the first one in the classpath.

**Comment by phxql** [comment]
> Hi @phxql . Are you still using this Liquibase Cache database extension?
> How about @onkar5891?
> @elktamer @ericfrancis @kurbatov @vivekmore?
> Any feedback?

I don't use it anymore.

**Comment by molivasdat** [comment]
@kurbatov Thanks for the feedback.  Are you still using H2 with Liquibase and are you still experiencing the issue?

**Comment by molivasdat** [comment]
Thanks @phxql for the feedback.

**Comment by kurbatov** [comment]
> @kurbatov Thanks for the feedback. Are you still using H2 with Liquibase and are you still experiencing the issue?

Yes, I do, in that project with monkey-patched `H2Database` class. It is not the latest version though. The issue exists in `3.8.7`. Tomorrow I'll check `3.10.1`.

**Comment by kurbatov** [comment]
It looks like the issue won't reproduce for H2 with `database_to_upper=false` for liqiuibase `3.10.1`.

**Comment by matze999** [comment]
I am still seeing the issue. Tried it with 3.10.1 and 4.0.0. I am running multiple spring.boot tests. The error i am seeing is:
`Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2020-07-22 07:56:51.187 ERROR [org.springframework.boot.SpringApplication] [Test worker:13]: Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.DatabaseException: Table "DATABASECHANGELOG" already exists; SQL statement:
CREATE TABLE PUBLIC.DATABASECHANGELOG (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10)) [42101-197] [Failed SQL: (42101) CREATE TABLE PUBLIC.DATABASECHANGELOG (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TIMESTAMP NOT NULL, ORDEREXECUTED INT NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10))]`

**Comment by matze999** [comment]
Hi, i can workaround the issue by changing the code a bit following @kurbatov suggestion. I implemented this for version liquibase version 4.0.0 and posted what i did over on stackoverflow (https://stackoverflow.com/questions/63036299/spring-boot-testing-with-liquibase-fails/63173456#63173456).

I do think that this is something which should be addressed in liquibase. Based on my understanding we should never fail if the "DATABASECHANGELOG" table creation fails because it already exists (there are legitimate reason to fail if the creation fails in general, but not due to it already existing).

Anyway, if my reasoning is sound, is there any way to work that into the official code?

**Comment by r2liquibase** [comment]
Hey, @molivasdat ! Do you need any help with this?

**Comment by molivasdat** [comment]
Hi @r2liquibase. Sure I'll take the help if you have a fix for this.

**Comment by r2liquibase** [comment]
Sounds like we need to override the method that deals with databasechangelog table creation. If this is realated to H2, can we get help from internal engineering?
